### PR TITLE
Slider: fix can not drag when step < 1

### LINKED ISSUE #1239: Slider 滑块拖动到边界卡死
### ElementUI version
 1.0.2 

### OS/Browers version
 win7/Chrome 53

### Vue version
2.0.3

Slider 滑块 step值为0.1，最小0，最大3.5，拖动到上边界3.5时卡死

另外step值为0.1，拖动的单位是1，而不是0.1，拖动的step应该和 step值保持一致，而现在不是

项目中在用Slider组件， 请帮修复，谢谢

### PR DESCRIPTION
**Pull request by furybean**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's [Contributing Guide](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.md).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

fixed #1239

### CODE REVIEW

**Comment by furybean** [comment]
@Leopoldthecoder

File: packages/slider/src/main.vue [hunk 1]
```diff
@@ -114,7 +115,11 @@
           this.$emit('input', this.max);
           return;
         }
-        this.inputValue = val;
+        if (val && this.precision) {
+          this.inputValue = parseFloat(val.toFixed(this.precision));
```
Comment (furybean) [inline packages/slider/src/main.vue]: Okay

File: packages/slider/src/main.vue [hunk 2]
```diff
@@ -114,6 +115,9 @@
           this.$emit('input', this.max);
           return;
         }
+        if (typeof val === 'number' && this.precision) {
+          val = parseFloat(val.toFixed(this.precision));
+        }
```
Comment (Leopoldthecoder) [inline packages/slider/src/main.vue]: 这样还是会出现我昨天说的问题😂
image

File: packages/slider/src/main.vue [hunk 3]
```diff
@@ -136,9 +140,9 @@
 
       setPosition(newPos) {
         if (newPos >= 0 && (newPos <= 100)) {
-          var lengthPerStep = 100 / ((this.max - this.min) / this.step);
-          var steps = Math.round(newPos / lengthPerStep);
-          this.$emit('input', Math.round(steps * lengthPerStep * (this.max - this.min) * 0.01 + this.min));
+          const lengthPerStep = 100 / ((this.max - this.min) / this.step);
+          const steps = Math.round(newPos / lengthPerStep);
+          this.$emit('input', steps * lengthPerStep * (this.max - this.min) * 0.01 + this.min);
```
Comment (furybean) [inline packages/slider/src/main.vue]: =.= 刚看到

**Pull request merged**